libjwt/jwt-gnutls.c

```diff
@@ -0,0 +1,231 @@
+/* Copyright (C) 2017 Nicolas Mora <mail@babelouest.org>
+   This file is part of the JWT C Library
+
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the JWT Library; if not, see
+   <http://www.gnu.org/licenses/>.  */
+
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+
+#include <gnutls/gnutls.h>
+#include <gnutls/crypto.h>
+#include <gnutls/x509.h>
+#include <gnutls/abstract.h>
+
+#include <jansson.h>
+
+#include <jwt.h>
+
+#include "jwt-private.h"
+#include "b64.h"
+#include "config.h"
+
+/**
+ * libjwt encryption/decryption function definitions
+ */
+int jwt_sign_sha_hmac(jwt_t *jwt, char **out, unsigned int *len, const char *str) {
+  int alg;
+  
+  switch (jwt->alg) {
+    case JWT_ALG_HS256:
+      alg = GNUTLS_DIG_SHA256;
+      break;
+    case JWT_ALG_HS384:
+      alg = GNUTLS_DIG_SHA384;
+      break;
+    case JWT_ALG_HS512:
+      alg = GNUTLS_DIG_SHA512;
+      break;
+    default:
```

Comment:
The default should be `return EINVAL`. No reason to set random data just to check it later. It's a failure.